### PR TITLE
refactor(async_worker): only store most recent bitcoin block header

### PR DESCRIPTION
**Pull request by arminsabouri**
This is actually the original design as defined in https://github.com/botanix-labs/botanix/blob/master/docs/pegin.md
User provides n block headers. The nth block header should be the tip that we store in reth. For v0 we only care about 1 conf. Going forward it will be m confs.

in a future PR. We will mine the block with the mth bitcoin block header in the botanix block header.

### CODE REVIEW

**Comment by stevenroose** [comment]
After https://github.com/botanix-labs/botanix/pull/178 this can just be block hash, right?

File: bin/reth/src/node/mod.rs [hunk 1]
```diff
@@ -272,10 +272,10 @@ impl<Ext: RethCliExt> NodeCommand<Ext> {
                         let block_hash = block_source.get_block_hash(current_tip).await.unwrap();
                         let block_header = block_source.get_block_header(block_hash).await.unwrap();
 
-                        header_write.push(block_header);
+                        // TODO (armins) in v1 we will need the nth deep block header not tip
+                        *header_write = Some(block_header);
```
Comment (stevenroose) [inline bin/reth/src/node/mod.rs:276]: You want the one that is 6 deep here, though, right? Instead of the tip..

**Review by stevenroose — CHANGES_REQUESTED** [review]
This seems to be already incorrect currently, but you want the 6-deep block hash right, not the current tip?

File: bin/reth/src/node/mod.rs (hunk 1 above)
Comment (arminsabouri) [reply bin/reth/src/node/mod.rs:276]: For v0 we want the tip. This is a product decision. V1 we will update that to 6 blocks deep

Comment (stevenroose) [reply bin/reth/src/node/mod.rs:276]: ah ok sorry didn't realize that. :+1:

**Review by stevenroose — APPROVED** [review]

File: bin/reth/src/node/mod.rs (hunk 1 above)
Comment (arminsabouri) [reply bin/reth/src/node/mod.rs:276]: All good All good 👍 I left a comment above it

**Pull request merged**